server/datastore/mysql/migrations/tables/20240228111134_AddScriptContentsTableAndRelationships.go

```diff
@@ -0,0 +1,190 @@
+		return fmt.Errorf("create table script_contents: %w", err)
+	}
+
+	insertScriptContentsStmt := `
+	INSERT INTO
+		script_contents (md5_checksum, contents)
+	VALUES
+		(UNHEX(?), ?)
+	ON DUPLICATE KEY UPDATE
+		created_at = created_at
+`
+
+	type scriptContent struct {
+		ID             uint   `db:"id"`
+		ScriptContents string `db:"script_contents"`
+	}
+
+	// load all contents found in host_script_results to next insert in
+	// script_contents
+	readHostScriptsStmt := `
+	SELECT
+		id,
+		script_contents
+	FROM
+		host_script_results
+`
+	var hostScripts []scriptContent
+	if err := txx.Select(&hostScripts, readHostScriptsStmt); err != nil {
+		return fmt.Errorf("load host_script_results contents: %w", err)
+	}
+
+	// load all contents found in scripts to next insert in script_contents
+	readScriptsStmt := `
+	SELECT
+		id,
+		script_contents
+	FROM
+		scripts
+`
+	var savedScripts []scriptContent
+	if err := txx.Select(&savedScripts, readScriptsStmt); err != nil {
+		return fmt.Errorf("load saved scripts contents: %w", err)
+	}
+
+	// for every script content, md5-hash it and keep track of the hash
+	// associated with the ids to update the host_script_results and scripts
+	// tables with the new script_content_id later on.
+	hostScriptsLookup := make(map[string][]uint, len(hostScripts))
+	savedScriptsLookup := make(map[string][]uint, len(savedScripts))
+	for _, s := range hostScripts {
+		hexChecksum := md5ChecksumScriptContent(s.ScriptContents)
+		hostScriptsLookup[hexChecksum] = append(hostScriptsLookup[hexChecksum], s.ID)
+		if _, err := txx.Exec(insertScriptContentsStmt, hexChecksum, s.ScriptContents); err != nil {
+			return fmt.Errorf("create script_contents from host_script_results: %w", err)
+		}
+	}
+	for _, s := range savedScripts {
+		hexChecksum := md5ChecksumScriptContent(s.ScriptContents)
+		savedScriptsLookup[hexChecksum] = append(savedScriptsLookup[hexChecksum], s.ID)
+		if _, err := txx.Exec(insertScriptContentsStmt, hexChecksum, s.ScriptContents); err != nil {
+			return fmt.Errorf("create script_contents from saved scripts: %w", err)
+		}
+	}
+
+	alterAddHostScriptsStmt := `
+ALTER TABLE host_script_results
+	ADD COLUMN script_content_id INT(10) UNSIGNED NULL,
+	ADD FOREIGN KEY (script_content_id) REFERENCES script_contents (id) ON DELETE CASCADE`
+	if _, err := tx.Exec(alterAddHostScriptsStmt); err != nil {
+		return fmt.Errorf("alter add column to table host_script_results: %w", err)
+	}
+
+	alterAddScriptsStmt := `
+ALTER TABLE scripts
+	ADD COLUMN script_content_id INT(10) UNSIGNED NULL,
+	ADD FOREIGN KEY (script_content_id) REFERENCES script_contents (id) ON DELETE CASCADE`
+	if _, err := tx.Exec(alterAddScriptsStmt); err != nil {
+		return fmt.Errorf("alter add column to table scripts: %w", err)
+	}
+
+	updateHostScriptsStmt := `
+UPDATE
+	host_script_results
+SET
+	script_content_id = (SELECT id FROM script_contents WHERE md5_checksum = UNHEX(?)),
+	updated_at = updated_at
```

Comment:
I keep the `updated_at` unmodified although it's not strictly required for `host_script_results` (we don't apply meaning to this timestamp AFAIK). Just a good habit I think, with our migrations.